db/sql/0_msar.sql

```diff
@@ -1590,6 +1740,318 @@ END;
 $$ LANGUAGE plpgsql;
 
 
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+-- COLUMN ALTERATION FUNCTIONS
+--
+-- Functions in this section should be related to altering columns' names, types, and constraints.
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+
+
+-- Rename columns ----------------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+__msar.rename_column(tab_name text, old_col_name text, new_col_name text) RETURNS text AS $$/*
+Change a column name, returning the command executed
+
+Args:
+  tab_name: The qualified, quoted name of the table where we'll change a column name
+  old_col_name: The quoted name of the column to change.
+  new_col_name: The quoted new name for the column.
+*/
+DECLARE
+  cmd_template text;
+BEGIN
+  cmd_template := 'ALTER TABLE %s RENAME COLUMN %s TO %s';
+  IF old_col_name <> new_col_name THEN
+    RETURN __msar.exec_ddl(cmd_template, tab_name, old_col_name, new_col_name);
+  ELSE
+    RETURN null;
+  END IF;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.rename_column(tab_id oid, col_id integer, new_col_name text) RETURNS smallint AS $$/*
+Change a column name, returning the command executed
+
+Args:
+  tab_id: The OID of the table whose column we're renaming
+  col_id: The ID of the column to rename
+  new_col_name: The unquoted new name for the column.
+*/
+BEGIN
+  PERFORM __msar.rename_column(
+    tab_name => __msar.get_relation_name(tab_id),
+    old_col_name => msar.get_column_name(tab_id, col_id),
+    new_col_name => quote_ident(new_col_name)
+  );
+  RETURN col_id;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION __msar.build_cast_expr(val text, type_ text) RETURNS text AS $$/*
+Build an expression for casting a column in Mathesar, returning the text of that expression.
+
+We fall back silently to default casting behavior if the mathesar_types namespace is missing.
+However, we do throw an error in cases where the schema exists, but the type casting function
+doesn't. This is assumed to be an error the user should know about.
+
+Args:
+  val: This is quite general, and isn't sanitized in any way. It can be either a literal or a column
+       identifier, since we want to be able to produce a casting expression in either case.
+  type_: This type name string must cast properly to a regtype.
+*/
+SELECT CASE
+  WHEN EXISTS (SELECT 1 FROM pg_namespace WHERE nspname='mathesar_types') THEN
```

Comment:
I recognize the `SELECT 1` trick from having to do the same, but the intent wouldn't be immediately obvious to someone else, possibly. Maybe move checking whether a schema exists to a function?

Reply:
Good point; done.